shared/fs/banner/public-reminder.tsx

```diff
@@ -0,0 +1,83 @@
+import * as React from 'react'
+import * as Kb from '../../common-adapters'
+import * as Types from '../../constants/types/fs'
+import * as Constants from '../../constants/fs'
+import openUrl from '../../util/open-url'
+import * as FsGen from '../../actions/fs-gen'
+import * as Container from '../../util/container'
+
+type OwnProps = {
+  path: Types.Path
+}
+
+type Props = {
+  hidden: boolean
+  onClose: () => void
+  onLoad: () => void
+  show: boolean
+  url: string
+}
+const PublicBanner = (props: Props) => {
+  if (props.show) {
+    return (
+      props.hidden && (
+        <Kb.Banner
+          color="yellow"
+          text="Everything you upload in here can be viewed by everyone at "
+          actions={[
+            {
+              onClick: () => openUrl(props.url),
+              title: props.url + '.',
+            },
+          ]}
+          onClose={props.onClose}
+        />
+      )
+    )
+  } else {
+    props.onLoad()
+    return null
+  }
+}
+
+type StateProps = {_lastClosedTlf: string; _writable: boolean}
+const mapStateToProps = (state, ownProps: OwnProps) => ({
+  _lastClosedTlf: state.fs.lastPublicBannerClosedTlf,
+  _writable: state.fs.pathItems.get(ownProps.path, Constants.unknownPathItem).writable,
+})
+
+type DispatchProps = {_onClose: (string) => () => void}
+const mapDispatchToProps = dispatch => ({
+  _onClose: tlf => () => dispatch(FsGen.createSetLastPublicBannerClosedTlf({tlf})),
```

Comment:
Could also compute `tlf` here from `ownProps`. I suppose you did it this way so `parsePath` is only called once? Perhaps we could memoize-one `parsePath`.

Reply:
also, reusing `_onClose` to clear the state. How about renaming `_onClose` to `setLastPublicBannerClosedTlf` or something?